blockreassurance.php

```diff
@@ -19,339 +19,418 @@
-                `id_reassurance` INT UNSIGNED NOT NULL AUTO_INCREMENT,
-                `id_lang` int(10) unsigned NOT NULL ,
-                `text` VARCHAR(300) NOT NULL,
-                PRIMARY KEY (`id_reassurance`, `id_lang`)
-            ) ENGINE='._MYSQL_ENGINE_.' DEFAULT CHARSET=utf8 ;');
-
-        return $return;
+        $this->_errors[] = $this->trans('There was an error during the installation. Please contact us through Addons website.', array(), 'Modules.Blockreassurance.Admin');
+        return false;
     }
 
+    /**
+     * uninstall module config()
+     *
+     * @return bool
+     */
     public function uninstall()
     {
-        return Configuration::deleteByName('BLOCKREASSURANCE_NBBLOCKS') &&
-            $this->uninstallDB() &&
-            parent::uninstall();
+        include(dirname(__FILE__).'/sql/uninstall.php');
+
+        Configuration::deleteByName('PSR_HOOK_HEADER');
+        Configuration::deleteByName('PSR_HOOK_FOOTER');
+        Configuration::deleteByName('PSR_HOOK_PRODUCT');
+        Configuration::deleteByName('PSR_HOOK_CHECKOUT');
+        Configuration::deleteByName('PSR_ICON_COLOR');
+        Configuration::deleteByName('PSR_TEXT_COLOR');
+
+        if (parent::uninstall()) {
+            return true;
+        }
+
+        $this->_errors[] = $this->trans('There was an error during the uninstallation. Please contact us through Addons website.', array(), 'Modules.Blockreassurance.Admin');
+        return false;
     }
 
-    public function uninstallDB()
+    /**
+     * load dependencies
+     */
+    public function loadAsset()
     {
-        return Db::getInstance()->execute('DROP TABLE IF EXISTS `'._DB_PREFIX_.'reassurance`') && Db::getInstance()->execute('DROP TABLE IF EXISTS `'._DB_PREFIX_.'reassurance_lang`');
+        $this->addJsDefList();
+
+        $controller = $this->context->controller;
+
+        $aCss = array(
+            $this->lib_path.'pickr/css/pickr.min.css',
+            $this->lib_path.'pickr/css/pickr-override.css',
+            $this->css_path.'/templates/display.css',
+            $this->css_path.'/templates/config.css',
+            $this->css_path.'/templates/listing.css',
+            $this->css_path.'/templates/reassurance_block.css',
+            $this->css_path.'/templates/appearance.css',
+            $this->css_path.'style.css',
+            $this->css_path.'faq.css',
+            $this->css_path.'menu.css',
+        );
+
+        $aJs = array(
+            $this->lib_path.'/pickr/js/pickr.js',
+            $this->js_path.'/appearance/colorpicker.js',
+            $this->js_path.'back.js',
+            $this->js_path.'menu.js',
+            $this->js_path.'vue.min.js',
+        );
+
+        $aCss[] = '//fonts.googleapis.com/icon?family=Material+Icons';
+
+        $this->context->controller->addCSS($aCss, 'all');
+        $this->context->controller->addJS($aJs);
+        $controller->addJqueryPlugin('colorpicker');
```

Comment:
If you use `$this->context->controller` you dont have to declare `$controller = $this->context->controller;` at line 205